extensions/ql-vscode/test/vscode-tests/cli-integration/jest.setup.ts

```diff
@@ -31,6 +31,25 @@ beforeAll(async () => {
   }
 
   await beforeAllAction();
+
+  // Activate the extension
+  const extension = await getActivatedExtension();
+
+  if (process.env.CLI_VERSION && process.env.CLI_VERSION !== "nightly") {
+    const cliVersion = await extension.cliServer.getVersion();
+
+    if (cliVersion.compare(process.env.CLI_VERSION) !== 0) {
+      // This calls the private `updateConfiguration` method in the `ConfigListener`
+      // It seems like the CUSTOM_CODEQL_PATH_SETTING.updateValue() call in
```

Comment:
Apologies for the ignorance, but since this is a problem with  `CUSTOM_CODEQL_PATH_SETTING.updateValue()`, why not trigger an update to the config after that instead of here? 

Also would it make sense to have a public `forceUpdateConfiguration()` (with comments etc. to say that ideally it would only be used by tests) instead of these castings?

Reply:
There is a possibility that the extension is not yet activated when we call `updateValue`, so we wouldn't be able to call this method there. By moving it here, we ensure that the extension is activated and the method can be called.

Adding a public `forceUpdateConfiguration` method makes sense, I've added that and added some comments to say that it should only be used in tests.